pilot/cmd/pilot-agent/config.go

```diff
@@ -66,7 +66,7 @@ func constructProxyConfig(role *model.Proxy) (meshconfig.ProxyConfig, error) {
 		if byResources != nil {
 			proxyConfig.Concurrency = byResources
 		}
-	} else {
+	} else if concurrency != 0 && proxyConfig.Concurrency == &types.Int32Value{Value: int32(0)} {
```

Comment:
This condition seems a bit suspicious to me. How is the user expected to use this? I don't see how it actually fixes the linked issue; this would require the user to set the `--concurrency` flag, not allow them to use the annotation?